### ISSUE TITLE
Product Price History Bug - Inventory Valuation

### ISSUE THREAD

**Issue by richard-willdooit** [odoomrp/odoomrp-wip#630]
Further to #627

Running  the "Current Inventory Valuation" with product price history installed calls  _get_inventory_value in stock_account.  This calls product_tmpl_obj.get_history_price, which gets a random variant price, not the template price necessarily, and not the price of the explicit variant.

As stated in #627, my understanding is it should get the explicit product's price.
But if what I stated there is wrong, then this code here should get the template's price.

Either way, this report is not correct.

**Comment by anajuaristi** [comment]
Yes... this is due Odoo core is not including cost_price on specific variants. Field is stored on template so with this aproach all variants from a template should have same price. 

This causes several lateral efects on product cost history, inventory valuation, manufacturing orders cost an so on.

To minimize a little bit those efects we added the field cost price on specific variant product form and we did a module that allows including different cost prices on variants. 

If we take in mind this change we could extend inventory valuation report, to include this field + template's one or ... just replace the template amount field and show the specific product one.

@pedrobaeza Opinion?

**Comment by richard-willdooit** [comment]
Yes, that is what I am getting at.  We installed the module to test the possibility of using it to value stock differently for each variant.  But this sort of depends on the other issue posted.

**Comment by anajuaristi** [comment]
We decided to include a new field on inventory valuation form and quants where we will calculate the cost using cost field on product.product form. 
We will not erase not rewrite the standar calculation valuation methods.
@agaldona

**Comment by anajuaristi** [comment]
@agaldona 
After improving the stimated and real cost modules, we should finish also this one.

As we said, we should add a new field on inventory valuation, without rewriting the standar calculation, where we show exactly same functionality but using the product.product form value when it's set to medium or standar price.

---

@agaldona 
Tal y como comentamos más arriba, la idea es incluir un nuevo campo en la pantalla de valoración de inventario y en la de quants, que realice exactamente lo mismo que el campo standar, pero, en vez de coger el campo precio coste de la plantilla que coja el del producto. Las condiciones deben ser iguales que el campo de odoo. Mostrará el valor cantidad \* campo cost del quant cuando el método es real y cantidad \* campo cost_price del product.product cuando el método es medio o standar.

Aparte de esto, considero imprescindible que hay que realizar
.- Mostrar en estos listados al lado del producto el método que se está usando
.- Mostrar en el listado de quants el campo cost

Con estas 2 nuevas cosas, al menos sabemos cómo está valorando ya que cualquiera de los 3 métodos puede ser aplicado a cualquier producto. 
También le metería una agrupación por método.

@pedrobaeza por fa, valida este enfoque

**Comment by pedrobaeza** [comment]
Yo no añadiría campos, si no que sustituiría cómo lo hacen los actuales. Tú tienes productos en los quants, no plantillas, por lo que no tiene sentido la valoración en precio plantilla.

**Comment by anajuaristi** [comment]
Entonces sí que sobreescribimos la forma standar en la que lo hace Odoo y dejamos solo la nuestra
Si dejamos las 2 mantenemos lo que viene de base y se puede comparar.
A mí me da un poco igual.

**Comment by pedrobaeza** [comment]
Yo prefiero sobreescribir para ser coherentes. Es además menos código, porque sólo es cambiar los métodos de cálculo. En pantalla no hay que poner nada más.

**Comment by anajuaristi** [comment]
Adelante asi entonces.
Como ves lo de mostrar related el metodo de calculo del producto en el tree
sin ir a ciegas?

**Comment by pedrobaeza** [comment]
No entiendo qué quieres hacer

**Comment by anajuaristi** [comment]
Nada... al lado del producto mostrar si el método de cálculo seleccionado
en la ficha de producto es estándar, medio o real. De esa forma puedes ver
cómo está calculando el total. Campo related, sin más. Y quizás meter un
filtro/agrupación por ese campo, pero tampoco sería imprescindible. Verlo
en la lista sí.
 Y mostrar el campo cost del quant, para que se vea el valor unitario
también.

2015-07-11 11:16 GMT+02:00 Pedro M. Baeza notifications@github.com:

> No entiendo qué quieres hacer
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/odoomrp/odoomrp-wip/issues/630#issuecomment-120597776
> .

## 

CEO Avanzosc, S.L http://www.avanzosc.es : Office phone / Tfono oficina:
(+34) 943 02 69 02
Ana Juaristi Olalde http://www.anajuaristi.com/: Personal phone: 677 93
42 59. User/usuario skype: Avanzosc
www.openerpsite.com

_El contenido de esta comunicación y de toda su documentación anexa es
confidencial y se dirige exclusivamente a su destinatario. El uso no
autorizado de esta información está prohibido por la legislación vigente.
Si usted no es el destinatario le rogamos nos lo indique, no comunique su
contenido a terceros y proceda a su destrucción. Disculpe las molestias que
le haya ocasionado la recepción indebida de este e-mail. Sus datos figuran
en un fichero cuyo titular es Avanzosc, S.L., a quien usted puede dirigirse
para ejercer sus derechos de acceso, rectificación, cancelación y oposición
en Klara Donea 13, 20720, Azkoitia (Gipuzkoa), Tef. 943 02 69 02 -
*_administracion@avanzosc.
soporte@avanzosc.eses*

_Komunikazio honen edukia eta dokumentazio erantsia konfidentziala da eta
hartzaileak bakarrik jaso beharko luke. Indarrean dagoen legeriak debekatu
egiten du bertan eskainitako informazioa baimenik gabe erabiltzea.
Komunikazioa zuri iritsi bazaizu, baina zu ez bazara hartzailea, mesedez,
guri jakinarazi, eta jasotako informazioa ez inori jakinarazi eta suntsitu.
Barkatu okerreko email hau jasotzeak eragindako eragozpenak. Zure datuak
Avanzosc, S.L. enpresaren fitxategietan sartuta daude. Zure datuak atzitzea
eska dezakezu, bai eta, datuak zuzentzea, ezereztea eta tratamenduari aurka
egitea ere. Horretarako, enpresara jo dezakezu, helbide honetan: *_Klara
Donea 13*\* 20720, Azkoitia (Gipuzkoa), telefonoa: 943 02 69 02 - *
_administracion@avanzosc. soporte@avanzosc.eses_ _This message and all
documents attached to it are confidential and intended only for the person
or entity to which it is addressed. Any use of this information by
unauthorised persons is prohibited under current legislation. If you
received this message by error, please advise us, destroy it and refrain
from communicating its contents to third parties. We apologise for any
inconvenience receiving this email improperly may cause to you. Your
personal data are included in a file owned by Avanzosc, S.L. If you want to
exercise your rights of access, correction, erasure and objection you can
contact the Controller at *_Klara Donea 13*\* 20720, Azkoitia (Gipuzkoa), T:
943 02 69 02 – administracion@avanzosc. soporte@avanzosc.eses*

**Comment by pedrobaeza** [comment]
No lo veo necesario, ya que se supone que tu método de cálculo va a ser homogéneo. Lo llevo diciendo mucho tiempo, que lo suyo sería mantener todos los tipos de precio en una tabla. dentro del producto.

**Comment by anajuaristi** [comment]
No es homogeneo. Se define en cada producto.

**Comment by pedrobaeza** [comment]
Digo que es homogéneo porque la contabilidad española exige que sea precio medio para cualquier valoración de inventario. Si tú colocas cualquier otro método, estás incumpliendo esa norma. De ahí mi insistencia continua de tener una tabla de precios de coste, parametrizable, no añadiendo campos...
